Sources/Swift/Tools/SentryLogBatcher.swift

```diff
@@ -6,14 +6,92 @@ import Foundation
 @_spi(Private) public class SentryLogBatcher: NSObject {
     
     private let client: SentryClient
+    private let flushTimeout: TimeInterval
+    private let maxBufferSize: Int
```

Comment:
Not a blocker, but do we have a limit on the log length size?
It isn't the same to have 100 logs with 100 bytes each, or 100 logs with a couple of kb (I imagine someone logging a codable model by mistake).

This buffer size seems to only limit the number of logs

Reply:
Yeah that sounds like a great approach. Will also change to either flushing after the 5 sec timeout (started when first log is added) or when the 1MB size limit of the data is exceeded.